scarb/src/ops/lockfile.rs

```diff
@@ -0,0 +1,16 @@
+use crate::core::lockfile::Lockfile;
+use crate::core::Workspace;
+use crate::internal::fsx;
+use anyhow::Result;
+
+#[tracing::instrument(skip_all, level = "debug")]
+pub fn read_lockfile(ws: &Workspace<'_>) -> Result<Lockfile> {
+    Lockfile::from_path(ws.lockfile_path())
+}
+
+#[tracing::instrument(skip_all, level = "debug")]
+pub fn write_lockfile(ws: &Workspace<'_>, lockfile: Lockfile) -> Result<()> {
+    let path = ws.lockfile_path();
+    fsx::write(path, lockfile.render()?.as_bytes())?;
```

Comment:
Wait... I think we should lock this file exclusively here to prevent corruption in case Scarb is executed multiple times in parallel... 🤔

Reply:
Good catch, thank you!

Reply:
Don't forget to also lock in shared mode in read op! 😉